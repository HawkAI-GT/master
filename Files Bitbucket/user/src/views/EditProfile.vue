<template>
  <v-card class="mx-auto" max-width="434" tile>
    <UserInfo v-bind:user="user" />
    <v-card-subtitle class="pb-0">Account Information</v-card-subtitle>
    <UserMetadata v-bind:user="user" />
    <UserCardActions />
  </v-card>
</template>

<script>
import UserMetadata from "../components/UserMetadata.vue";
import UserInfo from "../components/UserInfo.vue";
import UserCardActions from "../components/UserCardActions.vue";

export default {
  name: "EditProfile",

  components: {
    UserInfo,
    UserMetadata,
    UserCardActions
  },
  data() {
    return {
      user: {
        name: "Marcus Obrien",
        mall: "Oakland Mall",
        age: "26",
        email: "example@aja.com",
        dir: "My house",
        phone: "+502 3404-4860"
      }
    };
  }
};
</script>