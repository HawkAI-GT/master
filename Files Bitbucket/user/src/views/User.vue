<template>
  <v-card class="mx-auto" max-width="434" tile>
    <UserInfo v-bind:user="user" />
    <v-card-subtitle class="pb-0">Account Information</v-card-subtitle>
    <v-divider class="mx-4"></v-divider>
    <AccountInfo />
    <v-divider class="mx-4"></v-divider>
    <Support />
    <ButtonEdit />
  </v-card>
</template>

<script>
import UserInfo from "../components/UserInfo";
import ButtonEdit from "../components/ButtonEdit";
import AccountInfo from "../components/AccountInfo";
import Support from "../components/Support";

export default {
  name: "User",

  components: {
    UserInfo,
    ButtonEdit,
    AccountInfo,
    Support
  },
  data() {
    return {
      user: {
        name: "Marcus Obrien",
        mall: "Oakland Mall",
        age: "26",
        email: "example@aja.com",
        dir: "My house",
        phone: "+502 3404-4860"
      }
    };
  }
};
</script>