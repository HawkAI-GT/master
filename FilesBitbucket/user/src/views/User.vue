<template>
  <div>
    <v-card class="mx-auto" tile>
      <UserInfo v-bind:user="user" />
      <v-divider class="mx-4"></v-divider>
      <AccountInfo />
      <v-divider class="mx-4"></v-divider>
      <Support />
    </v-card>
    <BottomNav />
  </div>
</template>

<script>
import UserInfo from "../components/UserInfo";

import AccountInfo from "../components/AccountInfo";
import Support from "../components/Support";
import BottomNav from "../components/BottomNav";

export default {
  name: "User",

  components: {
    UserInfo,
    AccountInfo,
    Support,
    BottomNav
  },
  data() {
    return {
      user: {
        name: "Marcus Obrien",
        mall: "Oakland Mall",
        age: "26",
        email: "marcus@hawkai.me",
        dir: "My house",
        phone: "+502 3404-4860"
      }
    };
  }
};
</script>