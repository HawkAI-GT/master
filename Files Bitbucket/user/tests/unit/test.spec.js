import { mount } from '@vue/test-utils'
import User from '@/views/User.vue'
import EditProfile from '@/views/EditProfile.vue'

describe('Component', () => {
    // Now mount the component and you have the wrapper
    const wrapper = mount(User)

    it('renders the correct markup', () => {
        expect(wrapper.html()).toContain('<div class="v-card__subtitle pb-0">Account Information</div>')
    }),

        test('EditProfile renders correctly', () => {
            const wrapper = mount(EditProfile)
            expect(wrapper.element).toMatchSnapshot()
        })
})