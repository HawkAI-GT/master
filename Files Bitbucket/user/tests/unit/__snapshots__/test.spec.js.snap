// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component EditProfile renders correctly 1`] = `
<v-card
  class="mx-auto"
  max-width="434"
  tile=""
>
  <v-img
    height="100%"
    src="https://cdn.vuetifyjs.com/images/cards/server-room.jpg"
  >
    <v-row
      align="end"
      class="fill-height"
    >
      <v-col
        align-self="start"
        class="pa-0"
        cols="12"
      >
        <v-avatar
          class="profile"
          color="grey"
          size="164"
          tile=""
        >
          <v-img
            src="https://cdn.vuetifyjs.com/images/profiles/marcus.jpg"
          />
        </v-avatar>
      </v-col>
       
      <v-col
        class="py-0"
      >
        <v-list-item
          color="rgba(0, 0, 0, .4)"
          dark=""
        >
          <v-list-item-content>
            <v-list-item-title
              class="title"
            >
              Marcus Obrien
            </v-list-item-title>
             
            <v-list-item-subtitle>
              Oakland Mall
            </v-list-item-subtitle>
          </v-list-item-content>
        </v-list-item>
      </v-col>
    </v-row>
  </v-img>
   
  <v-card-subtitle
    class="pb-0"
  >
    Account Information
  </v-card-subtitle>
   
  <v-card-text
    class="text--primary"
  >
    <div>
      <v-text-field
        label="E-mail"
        prepend-icon="mdi-email"
      />
       
      <v-text-field
        label="Phone"
        prepend-icon="mdi-phone"
      />
       
      <v-text-field
        label="Direction"
        prepend-icon="mdi-map-marker"
      />
       
      <v-text-field
        label="Age"
        prepend-icon="mdi-chart-donut"
      />
       
      <v-autocomplete
        items="Male,Female,Other"
        label="Sex"
        placeholder="Male"
        prepend-icon="mdi-gender-male-female"
        required=""
        rules="function () { return !!_vm.sex || 'This field is required'; }"
      />
    </div>
  </v-card-text>
   
  <v-card-actions>
    <v-btn
      color="red"
      outlined=""
      text=""
      to="/user"
    >
      Cancel
    </v-btn>
     
    <v-spacer />
     
    <v-btn
      color="green"
      outlined=""
      text=""
      to="/user"
    >
      Save
    </v-btn>
  </v-card-actions>
</v-card>
`;
